Terraform import fails for azurerm_iothub_dps

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

Terraform v0.12.21

Initializing provider plugins...
- Checking for available provider plugins...
- Downloading plugin for provider "azuread" (hashicorp/azuread) 0.7.0...
- Downloading plugin for provider "null" (hashicorp/null) 2.1.2...
- Downloading plugin for provider "external" (hashicorp/external) 1.2.0...
- Downloading plugin for provider "azurerm" (hashicorp/azurerm) 1.44.0...

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* `azurerm_iothub_dps`

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
resource "azurerm_iothub_dps" "iot_dps" {
  name                = local.dps_name
  resource_group_name = local.dps_resource_group
  location            = local.dps_location

  sku {
    name     = "S1"
    capacity = "1"
  }

  linked_hub {
    connection_string = data.external.connection_string.result["connection"]
    location          = azurerm_iothub.iot.location
  }

  count = var.location == "eastus2" ? 1 : 0
}
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

terraform  import -var stack=iot -var location=eastus2 -var-file=variables/prod.tfvars  azurerm_iothub_dps.iot_dps[0] /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/Cloud.Prod.cloud-iot/providers/Microsoft.Devices/ProvisioningServices/prod-cloud-iot-dps
azurerm_iothub_dps.iot_dps[0]: Importing from ID "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/Cloud.Prod.cloud-iot/providers/Microsoft.Devices/ProvisioningServices/prod-cloud-iot-dps"...

azurerm_iothub_dps.iot_dps[0]: Import prepared!
  Prepared azurerm_iothub_dps for import
azurerm_iothub_dps.iot_dps[0]: Refreshing state... [id=/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/Cloud.Prod.cloud-iot/providers/Microsoft.Devices/ProvisioningServices/prod-cloud-iot-dps]

Error: rpc error: code = Unavailable desc = transport is closing


panic: runtime error: invalid memory address or nil pointer dereference
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3212d76]
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4:
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: goroutine 79 [running]:
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/iothub.flattenIoTHubDPSSku(...)
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/iothub/resource_arm_iothub_dps.go:361
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/iothub.resourceArmIotHubDPSRead(0xc0005f0620, 0x39652e0, 0xc0000c86c0, 0x0, 0x0)
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/iothub/resource_arm_iothub_dps.go:250 +0x586
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000503900, 0xc0008d8280, 0x39652e0, 0xc0000c86c0, 0xc000148b10, 0x0, 0x0)
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:455 +0x119
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadResource(0xc00000eb08, 0x46b9440, 0xc000942360, 0xc0008d80a0, 0xc00000eb08, 0xc000942360, 0xc000991bd0)
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:525 +0x3d7
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadResource_Handler(0x3ccf080, 0xc00000eb08, 0x46b9440, 0xc000942360, 0xc000d9a060, 0x0, 0x46b9440, 0xc000942360, 0xc000750480, 0x178)
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3153 +0x23e
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0003c4580, 0x4710940, 0xc000092780, 0xc000d46100, 0xc000869e00, 0x7aa6730, 0x0, 0x0, 0x0)
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:995 +0x466
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: google.golang.org/grpc.(*Server).handleStream(0xc0003c4580, 0x4710940, 0xc000092780, 0xc000d46100, 0x0)
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1275 +0xda6
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000286250, 0xc0003c4580, 0x4710940, 0xc000092780, 0xc000d46100)
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:710 +0x9f
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-02-29T10:43:33.883-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:708 +0xa1
2020-02-29T10:43:33.892-0500 [DEBUG] plugin: plugin process exited: path=/home/mark/projects/devops/devops-cloud/iot/.terraform/plugins/linux_amd64/terraform-provider-azurerm_v1.44.0_x4 pid=26889 error="exit status 2"
2020/02/29 10:43:33 [ERROR] <root>: eval: *terraform.EvalRefresh, err: rpc error: code = Unavailable desc = transport is closing
2020/02/29 10:43:33 [ERROR] <root>: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020/02/29 10:43:33 [TRACE] [walkImport] Exiting eval tree: import azurerm_iothub_dps.iot_dps[0] result
2020/02/29 10:43:33 [TRACE] vertex "import azurerm_iothub_dps.iot_dps[0] result": visit complete
2020/02/29 10:43:33 [TRACE] vertex "azurerm_iothub_dps.iot_dps[0] (import id \"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/Cloud.Prod.cloud-iot/providers/Microsoft.Devices/ProvisioningServices/prod-cloud-iot-dps\")": dynamic subgraph encountered errors
2020/02/29 10:43:33 [TRACE] vertex "azurerm_iothub_dps.iot_dps[0] (import id \"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/Cloud.Prod.cloud-iot/providers/Microsoft.Devices/ProvisioningServices/prod-cloud-iot-dps\")": visit complete
2020/02/29 10:43:33 [TRACE] dag/walk: upstream of "provider.azurerm (close)" errored, so skipping
2020/02/29 10:43:33 [TRACE] dag/walk: upstream of "root" errored, so skipping
2020-02-29T10:43:34.123-0500 [DEBUG] plugin: plugin exited



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

SECURITY WARNING: the "crash.log" file that was created may contain
sensitive information that must be redacted before it is safe to share
on the issue tracker.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

### Expected Behavior

<!--- What should have happened? --->

### Actual Behavior

<!--- What actually happened? --->

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform  import -var stack=iot -var location=eastus2 -var-file=variables/prod.tfvars  azurerm_iothub_dps.iot_dps[0] /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/Cloud.Prod.cloud-iot/providers/Microsoft.Devices/ProvisioningServices/prod-cloud-iot-dps
azurerm_iothub_dps.iot_dps[0]: Importing from ID "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/Cloud.Prod.cloud-iot/providers/Microsoft.Devices/ProvisioningServices/prod-cloud-iot-dps"`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in a Azure China/Germany/Government? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Such as vendor documentation?
--->

* #0000

--- a/azurerm/internal/services/iothub/resource_arm_iothub_dps.go
+++ b/azurerm/internal/services/iothub/resource_arm_iothub_dps.go
@@ -219,6 +219,10 @@ func resourceArmIotHubDPSRead(d *schema.ResourceData, meta interface{}) error {
 	}
 	resourceGroup := id.ResourceGroup
 	name := id.Path["provisioningServices"]
+	// the name path can use the ProvisioningServices in older iterations
+	if name == "" {
+		name = id.Path["ProvisioningServices"]
+	}
 
 	resp, err := client.Get(ctx, name, resourceGroup)
 	if err != nil {

